File: src/main/resources/output/IFC4X3_java/Types/IfcTextFontSelect.java
// SELECT TYPE IfcTextFontSelect
public class IfcTextFontSelect {
    public Object value;

    public IfcTextFontSelect(Object value) {
        if (!(value instanceof IfcExternallyDefinedTextFont || value instanceof IfcPreDefinedTextFont)) {
            throw new IllegalArgumentException("Value must be one of: IfcExternallyDefinedTextFont, IfcPreDefinedTextFont");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}

File: src/main/resources/output/IFC4X3_js/Types/IfcTextFontSelect.js
// SELECT TYPE IfcTextFontSelect
// Options:
// - IfcExternallyDefinedTextFont
// - IfcPreDefinedTextFont
class IfcTextFontSelect {
    constructor(value) {
        if (!(value instanceof IfcExternallyDefinedTextFont || value instanceof IfcPreDefinedTextFont)) {
            throw new Error("Value must be one of: IfcExternallyDefinedTextFont, IfcPreDefinedTextFont");
        }
        this.value = value;
    }

    getValueAs(TypeClass) {
        if (this.value instanceof TypeClass) {
            return this.value;
        } else {
            throw new Error("Stored value is not of the requested type");
        }
    }
}
